pkg/kubelet/remote/remote_runtime.go

```diff
@@ -209,7 +209,11 @@ func (r *RemoteRuntimeService) StartContainer(containerID string) error {
 
 // StopContainer stops a running container with a grace period (i.e., timeout).
 func (r *RemoteRuntimeService) StopContainer(containerID string, timeout int64) error {
-	ctx, cancel := getContextWithTimeout(r.timeout)
+	ctx, cancel := getContextWithTimeout(time.Duration(timeout) * time.Second)
+	if timeout == 0 {
+		// Do not set timeout when timeout is 0.
+		ctx, cancel = getContextWithCancel()
```

Comment:
Shouldn't the logic is if timeout==0 for StopContainer, we set the default timeout value? The default one can be r.timeout here